### PR TITLE
Add support for Composer 2

### PR DESCRIPTION
**Pull request by maxime-rainville**
# Parent issue
* https://github.com/silverstripe/recipe-plugin/issues/15

### CODE REVIEW

**Comment by maxime-rainville** [comment]
Probably need to update the unit test so we try this thing with both compose 1 and composer 2.

File: src/VendorPlugin.php [hunk 1]
```diff
@@ -248,4 +248,14 @@ protected function installLibrary(IOInterface $IO, Library $library)
         );
         $task->process($IO, [$library]);
     }
+
+    public function deactivate(Composer $composer, IOInterface $io)
+    {
+
```
Comment (maxime-rainville) [inline src/VendorPlugin.php:254]:
```suggestion
```

File: src/VendorPlugin.php [hunk 2]
```diff
@@ -248,4 +248,14 @@ protected function installLibrary(IOInterface $IO, Library $library)
...
+    }
+
+    public function uninstall(Composer $composer, IOInterface $io)
+    {
+
```
Comment (maxime-rainville) [inline src/VendorPlugin.php:259]:
```suggestion
```

File: composer.json [hunk 3]
```diff
@@ -27,11 +27,11 @@
     },
     "minimum-stability": "dev",
     "require": {
-        "composer/installers": "^1.4",
-        "composer-plugin-api": "^1.1"
+        "composer/installers": "^1.4|^2",
```
Comment (dhensby) [inline composer.json:30]: the correct declaration is `^1.4 || ^2` (see https://getcomposer.org/doc/articles/versions.md#version-range)

File: .travis.yml [hunk 4]
```diff
@@ -18,7 +22,7 @@ before_script:
   - export PATH=~/.composer/vendor/bin:$PATH
   - composer validate
   - composer global require squizlabs/php_codesniffer:^3 --prefer-dist --no-interaction --no-progress --no-suggest -o
-  - composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
+  - composer install COMPOSER_FLAG --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
```
Comment (dhensby) [inline .travis.yml:25]: Doesn't this need to be "${COMPOSER_FLAG}"

File: .travis.yml [hunk 5]
```diff
@@ -18,7 +22,7 @@ before_script:
   - export PATH=~/.composer/vendor/bin:$PATH
   - composer validate
   - composer global require squizlabs/php_codesniffer:^3 --prefer-dist --no-interaction --no-progress --no-suggest -o
-  - composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
+  - composer update $COMPOSER_FLAG --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
```
Comment (dhensby) [inline .travis.yml:25]: `composer install` is correct, no? No other installs have happened and, if there's a lock file, we should really be using it?

Comment (sminnee) [reply .travis.yml:25]: --prefer-lowest doesn't support it. And I tend to think "without a composer.lock, install fails over to update, but update is what's actually happening"

**Comment by brynwhyman** [comment]
It looks like we should close this for https://github.com/silverstripe/vendor-plugin/pull/36 which has green tests?

**Comment by brynwhyman** [comment]
Should we close this for https://github.com/silverstripe/vendor-plugin/pull/36 which has green tests?

**Pull request closed without merging**